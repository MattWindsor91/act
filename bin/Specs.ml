(* This file is part of 'act'.

   Copyright (c) 2018 by Matt Windsor

   Permission is hereby granted, free of charge, to any person
   obtaining a copy of this software and associated documentation
   files (the "Software"), to deal in the Software without
   restriction, including without limitation the rights to use, copy,
   modify, merge, publish, distribute, sublicense, and/or sell copies
   of the Software, and to permit persons to whom the Software is
   furnished to do so, subject to the following conditions:

   The above copyright notice and this permission notice shall be
   included in all copies or substantial portions of the Software.

   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
   EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
   SOFTWARE. *)

open Core
open Lib

let command =
  let open Command.Let_syntax in
  Command.basic
    ~summary:"outputs information about the current compiler specs"
    [%map_open
      let standard_args = Standard_args.get in
      fun () ->
        Common.lift_command standard_args
          ~with_compiler_tests:false
          ~f:(fun _o cfg ->
              let specs = Config.M.compilers cfg in
              let verbose = Standard_args.is_verbose standard_args in
              Compiler.Spec.Set.pp_verbose verbose Format.std_formatter specs;
              Format.print_newline ();
              Result.ok_unit
            )
    ]
;;
