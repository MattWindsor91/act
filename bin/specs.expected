gcc.x86:
 Style: gcc
 Emits: x86.att
 Command: gcc -m32 -DNO_PTHREADS
 Machine: default (local)
 Herd: yes
